From 1b235305d35c3adc7a0d1bbc3943fac1ef84aab6 Mon Sep 17 00:00:00 2001
From: "A. Wilcox" <AWilcox@Wilcox-Tech.com>
Date: Thu, 14 Sep 2017 15:50:52 -0500
Subject: [PATCH] __lookup_serv: return correct error code

If AI_NUMERICSERV is specified and a numeric service was not provided,
POSIX mandates getaddrinfo return EAI_NONAME.  EAI_SERVICE is only for
services that cannot be used on the specified socket type.
---
 src/network/lookup_serv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/lookup_serv.c b/src/network/lookup_serv.c
index 66ebaea..403b12a 100644
--- a/src/network/lookup_serv.c
+++ b/src/network/lookup_serv.c
@@ -64,7 +64,7 @@ int __lookup_serv(struct service buf[static MAXSERVS], const char *name, int pro
 		return cnt;
 	}
 
-	if (flags & AI_NUMERICSERV) return EAI_SERVICE;
+	if (flags & AI_NUMERICSERV) return EAI_NONAME;
 
 	size_t l = strlen(name);
 
-- 
2.10.0

